### PR TITLE
fix: ssl need to be update when secret has been changed

### PR DESCRIPTION
**Pull request by gxthrj**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#324

### CODE REVIEW

File: test/e2e/ingress/secret.go [hunk 1]
```diff
@@ -144,6 +144,6 @@ RU+QPRECgYB6XW24EI5+w3STbpnc6VoTS+sy9I9abTJPYo9LpCJwfMYc9Tg9Cx2K
 		tlsUpdate, err := s.ListApisixTls()
 		assert.Nil(ginkgo.GinkgoT(), err, "list tlsUpdate error")
 		assert.Len(ginkgo.GinkgoT(), tlsUpdate, 1, "tls number not expect")
-		assert.NotEqual(ginkgo.GinkgoT(), KeyInApisix, tlsUpdate[0].Key, "tls key not expect")
+		assert.Equal(ginkgo.GinkgoT(), certUpdate, tlsUpdate[0].Cert, "tls cert not expect")
```
Comment (tokers) [inline test/e2e/ingress/secret.go:147]: Both the cert and key should be checked.

Comment (tokers) [inline test/e2e/ingress/secret.go:147]: We may also add some test cases to cover the ApisixTls delete.

Comment (gxthrj) [reply test/e2e/ingress/secret.go:151]: Done

Comment (gxthrj) [reply test/e2e/ingress/secret.go:151]: Done

**Comment by tokers** [comment]
@gxthrj Now we only verify that data are passed to the DP but the verifications of the use of these certificates are missing.

**Comment by gxthrj** [comment]
> @gxthrj Now we only verify that data are passed to the DP but the verifications of the use of these certificates are missing.

Yes, you are right, I will add it.

File: test/e2e/scaffold/scaffold.go [hunk 2]
```diff
@@ -175,6 +179,17 @@ func (s *Scaffold) NewAPISIXHttpsClient() *httpexpect.Expect {
 					// accept any certificate; for testing only!
 					InsecureSkipVerify: true,
 				},
+				DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
```
Comment (tokers) [inline test/e2e/scaffold/scaffold.go:182]: Why not set the `ServerName` in `tls.Config`?

Comment (gxthrj) [reply test/e2e/scaffold/scaffold.go:182]: Great, use `ServerName` instead.

File: test/e2e/ingress/secret.go [hunk 3]
```diff
@@ -89,61 +159,129 @@ wrw7im4TNSAdwVX4Y1F4svJ2as5SJn5QYGAzXDixNuwzXYrpP9rzA2s=
...
+		// check DP
+		s.NewAPISIXHttpsClient(host).GET("/ip").WithHeader("Host", host).Expect().Status(http.StatusOK).Body().Raw()
+
+		dialer := &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}
+
+		http.DefaultTransport.(*http.Transport).DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
```
Comment (tokers) [inline test/e2e/ingress/secret.go:174]: I think `DialContext` can also be removed if we already have the correct `ServerName`.

Comment (gxthrj) [reply test/e2e/ingress/secret.go:174]: Yes, dirty codes.

**Review by tokers — APPROVED** [review]

**Pull request merged**